scripts/scraper-ng/index.js

```diff
@@ -1,51 +1,79 @@
+const fetch = require("node-fetch");
 const rehype = require("rehype");
 const reporter = require("vfile-reporter");
 
+const mdnUrl = require("./mdn-url");
 const toVFile = require("./url-to-vfile");
 
 const examplePage =
   "https://developer.mozilla.org/en-US/docs/Web/HTML/Element/div";
+const exampleShorthand = "/en-US/docs/Web/HTML/Element/div";
 
 const argv = require("yargs")
-  .usage("Usage: $0 <url> …")
-  .example(`$0 ${examplePage}`, "scrape a page")
-  .example(`$0 -n ${examplePage}`, "dry run (lint)")
+  .parserConfiguration({ "boolean-negation": false })
+  .usage("Usage: $0 <url>")
+  .example(`$0 ${examplePage}`, "scrape a page and all its subpages")
+  .example(`$0 ${exampleShorthand}`, "omit the protocol and domain")
+  .example(`$0 --no-subpages ${exampleShorthand}`, "one page only")
+  .example(`$0 -n ${exampleShorthand}`, "dry run (lint)")
   .demandCommand(1, 1, "A URL is required", "Too many arguments")
+  .describe("no-subpages", "Don't walk the entire tree from the URL")
+  .default("no-subpages", false)
   .describe("n", "Dry run (lint)")
   .alias("n", "dry-run")
   .describe("q", "Suppress success messages")
   .alias("q", "quiet")
   .describe("v", "Expand linter notes, if applicable")
   .alias("v", "verbose")
-  .boolean(["n", "q", "v"]).argv;
+  .boolean(["n", "q", "v", "no-subpages"]).argv;
 
 const processor = rehype()
   .use([require("./plugins/kumascript-macros")])
   .use([require("./preset")]);
 // TODO: add YAML frontmatter insertion
 
 async function run() {
-  const vfiles = [];
+  const root = await fetchTree(argv._[0]);
+  const urls = argv.noSubpages ? [root.url] : flattenTree(root);
+  const toProcess = urls.map(toVFile);
+  const processed = [];
 
-  const file = await toVFile(argv._[0]);
+  for (const f of toProcess) {
+    const file = await f;
```

Comment:
This turned on a light bulb. Async is hard to reason about! 

My aim here was to start processing the HTML as soon as it had loaded for each file individually, rather than waiting for all of the HTML to have been loaded for all of the vfiles. But because I'm awaiting the promises sequentially, one slow page holds up the whole sequence anyway.

The thing that would actually make a difference would be to process each file when it loads. I'm pushing a change so that this happens. It also avoids some needless iteration over the to-be-processed files.